File: utils/TextUtils.go
package utils

import (
	"errors"
	"regexp"
)

func TextIsEmpty(text string) bool {
	return text == ""
}

func IsEmailValid(email string) bool {
	if TextIsEmpty(email) {
		return false
	}
	// Regular expression pattern for email validation
	pattern := `^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$`

	// Compile the pattern into a regular expression object
	regExp := regexp.MustCompile(pattern)

	// Match the email against the regular expression
	return regExp.MatchString(email)
}

// IsPasswordValid
// Password should be of 8 characters long
// Password should contain atleast one lower case character
// Password should contain at least one upper case character
// Password should contain atleast one digit
// Password should contain at least one special character
func IsPasswordValid(password string) error {
	if len(password) < 8 {
		return errors.New("password should be of 8 characters long")
	}
	done, err := regexp.MatchString("([a-z])+", password)
	if err != nil {
		return err
	}
	if !done {
		return errors.New("password should contain atleast one lower case character")
	}
	done, err = regexp.MatchString("([A-Z])+", password)
	if err != nil {
		return err
	}
	if !done {
		return errors.New("password should contain at least one upper case character")
	}
	done, err = regexp.MatchString("([0-9])+", password)
	if err != nil {
		return err
	}
	if !done {
		return errors.New("password should contain atleast one digit")
	}

	done, err = regexp.MatchString("([!@#$%^&*.?-])+", password)
	if err != nil {
		return err
	}
	if !done {
		return errors.New("password should contain at least one special character")
	}
	return nil
}

File: ui/src/Utils/TextUtils.ts
export default class TextUtils {
    public static isEmpty(text?: string | null) {
        return !text || text.trim() === ""
    }

    public static isNotEmpty(text?: string | null) {
        return !TextUtils.isEmpty(text)
    }

    public static valueOrEmpty(text: string | null | undefined, defaultV: string): string {
        return TextUtils.isEmpty(text) ? defaultV : text!
    }

    public static isEmailValid(email: string): boolean {
        if (!email.trim()) {
            return false
        }

        // Regular expression pattern for email validation
        const pattern = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/

        // Match the email against the regular expression
        return pattern.test(email)
    }

    public static PasswordError(password: string): string {
        let error = ""
        if (password.length < 8) {
            error = "Password should be at least 8 characters long"
        }

        let done = /[a-z]+/.test(password)
        if (!done) {
            error = "Password should contain at least one lowercase character"
        }

        done = /[A-Z]+/.test(password)
        if (!done) {
            error = "Password should contain at least one uppercase character"
        }

        done = /[0-9]+/.test(password)
        if (!done) {
            error = "Password should contain at least one digit"
        }

        done = /[!@#$%^&*.?-]+/.test(password)
        if (!done) {
            error = "Password should contain at least one special character"
        }

        return error
    }
}
